onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -347,6 +347,12 @@ def list(self, request, *args, **kwargs):
             tags = self.request.query_params.get('tags')
             not_tagged = self.request.query_params.get('not_tagged')
 
+            self.object_list = filters.InstanceFilter(
```

Comment:
DataViewSet is set up to deal primarily with XForm objects.  If we added InstanceFilter to the filter_backends property, then it would filter on XForm objects, which is not what we want.

The Instance objects are dealt with in the list method of DataViewSet - which is also the same place where Data filters are implemented, and this is where (and why) we are adding InstanceFilter at that point.